src/components/MapView/Download/reportDoc.tsx

```diff
@@ -0,0 +1,525 @@
+      marginBottom: 10,
+    },
+    title: {
+      fontSize: 10.78,
+      fontWeight: 500,
+    },
+    subText: {
+      fontSize: 9.24,
+    },
+    mapImage: {
+      width: '100%',
+      maxHeight: 284,
+    },
+    legendsContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      justifyContent: 'flex-start',
+      backgroundColor: theme.pdf?.legendsBackgroundColor,
+    },
+    legend: {
+      height: 130,
+      display: 'flex',
+      flexDirection: 'column',
+      marginRight: 10,
+      padding: 10,
+    },
+    legendTittle: {
+      fontSize: 10.78,
+      paddingBottom: 10,
+      fontWeight: 500,
+    },
+    legendContentsWrapper: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexWrap: 'wrap',
+      height: '100%',
+    },
+    legendContent: {
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      fontSize: 10.78,
+      paddingTop: 5,
+      paddingRight: 5,
+    },
+    legendText: {
+      paddingLeft: 10,
+    },
+    dash: {
+      width: 16,
+      height: 2,
+    },
+    borderedBox: {
+      width: 12,
+      height: 12,
+      border: 2,
+      borderRadius: 3,
+    },
+    box: {
+      width: 10,
+      height: 10,
+    },
+    tableName: {
+      padding: 6,
+      fontWeight: 500,
+    },
+    tableCell: {
+      padding: 8,
+      fontSize: 9.24,
+    },
+    tableRow: {
+      display: 'flex',
+      flexDirection: 'row',
+      lineHeight: 1,
+    },
+    tableHead: {
+      borderBottom: 1,
+      borderBottomColor: theme.pdf?.table?.borderColor,
+      backgroundColor: theme.pdf?.table?.darkRowColor,
+      fontWeight: 500,
+    },
+    tableFooter: {
+      borderTop: 1,
+      borderTopColor: theme.pdf?.table?.borderColor,
+      fontWeight: 700,
+    },
+    footer: {
+      position: 'absolute',
+      fontSize: 9.24,
+      bottom: 10,
+      left: 10,
+      right: 0,
+      textAlign: 'left',
+    },
+  });
+
+interface LegendProps {
+  theme: Theme;
+}
+
+const AreasLegend = ({ theme }: LegendProps) => {
+  const styles = makeStyles(theme);
+  return (
+    <View style={styles.legend}>
+      <View>
+        <Text style={styles.legendTittle}>Areas</Text>
+      </View>
+      <View style={styles.legendContentsWrapper}>
+        <View style={styles.legendContent}>
+          <View style={[styles.dash, { backgroundColor: '#000000' }]} />
+          <Text style={[styles.legendText]}>Province</Text>
+        </View>
+        <View style={styles.legendContent}>
+          <View style={[styles.dash, { backgroundColor: '#999797' }]} />
+          <Text style={[styles.legendText]}>District</Text>
+        </View>
+        <View style={styles.legendContent}>
+          <View style={[styles.dash, { backgroundColor: '#D8D6D6' }]} />
+          <Text style={[styles.legendText]}>Township</Text>
+        </View>
+      </View>
+    </View>
+  );
+};
+
+const StormWindBuffersLegend = ({ theme }: LegendProps) => {
```

Comment:
Similar code already exists to generate the legend, let's try to make this more programable and future proof.
Suggestions:
1. Let's see if we can re-use some of the existing code from `Legends`
2. If not, let's make this an object/list and use a `.map` to avoid repeating styling and all so many times.

Especially true for the flood legend which already exists as an object in the analysisData.hazardLayer.legend IIRC